### ISSUE TITLE
Authorization fails for specific domains

### ISSUE THREAD

**Issue by jennedebleser** [Neilpang/acme.sh#1936]
TL;DR It fails because of a unescaped grep string.

Steps to reproduce
------------------
1. Request a certificate for the domains: api.us.example.com and api-us.example.com
2. Authorization fails for 1 domain

Debug log
-----------------

```
acme.sh  --issue -d api.us.example.com -d api-us.example.com --debug 2
```

```
[Tue Nov 27 13:09:06 UTC 2018] d='*.api.us.example.com'
[Tue Nov 27 13:09:06 UTC 2018] Getting webroot for domain='*.api.us.example.com'
[Tue Nov 27 13:09:06 UTC 2018] _w='dns_da'
[Tue Nov 27 13:09:06 UTC 2018] _currentRoot='dns_da'
```
Notice this part contains 2 json objects from the $_authorizations_map
```
[Tue Nov 27 13:09:06 UTC 2018] response='{"identifier":{"type":"dns","value":"api.us.example.com"},"status":"pending","expires":"2018-12-04T11:00:27Z","challenges":[{"type":"dns-01","status":"pending","url":"https://acme-v02.api.letsencrypt.org/acme/REDACTED","token":"REDACTED"}],"wildcard": true}
{"identifier":{"type":"dns","value":"api-us.example.com"},"status":"valid","expires":"2018-12-27T11:03:10Z","challenges":[{"type":"dns-01","status":"valid","url":"https://acme-v02.api.letsencrypt.org/acme/challenge/REDACTED","token":"REDACTED","validationRecord":[{"hostname":"api-us.example.com"}]}],"wildcard": true}'
```

```
[Tue Nov 27 13:09:06 UTC 2018] entry='"type":"dns-01","status":"pending","url":"https://acme-v02.api.letsencrypt.org/acme/challenge/REDACTED","token":"REDACTED"
"type":"dns-01","status":"valid","url":"https://acme-v02.api.letsencrypt.org/acme/challenge/REDACTED","token":"REDACTED","validationRecord":[{"hostname":"api-us.example.com"'
[Tue Nov 27 13:09:06 UTC 2018] token='REDACTED'
[Tue Nov 27 13:09:06 UTC 2018] uri='https://acme-v02.api.letsencrypt.org/acme/challenge/REDACTED'
[Tue Nov 27 13:09:06 UTC 2018] keyauthorization='REDACTED'
[Tue Nov 27 13:09:06 UTC 2018] *.api.us.example.com is already verified.
[Tue Nov 27 13:09:07 UTC 2018] keyauthorization='verified_ok'
[Tue Nov 27 13:09:07 UTC 2018] dvlist='*.api.us.example.com#verified_ok#https://acme-v02.api.letsencrypt.org/acme/challenge/REDACTED#dns-01#dns_da'
```

Problem located at https://github.com/Neilpang/acme.sh/blob/393d8b9ded7299b5ecbec05a71fe203d94fdb030/acme.sh#L3692
`grep "^$d,"` becomes `grep "^api.us.example.com,"` where the dots are special regex characters that indicate it can match any character which causes both domains to match from `$_authorizations_map` resulting in the detection of the "valid" status from the other domain.

Solution
-----------------
Escape the `$d` param before passing it to grep:
```bash
d=$(sed 's/[]\.|$(){}?+*^]/\\&/g' <<< $d)
```

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.

**Comment by neilpang** [comment]
Fixed in d04c6dd3 (refined in 3761fb43): multiple grep matches in the authorization map are now disambiguated with a literal per-character comparison, so api.us.example.com no longer matches api-us.example.com.
